### DISCUSSION TITLE
How do I resolve "cannot import module "greeter" (not found)"

### DISCUSSION THREAD

**Discussion by jason-pezzimenti** [vlang/v#21979] [Questions and Answers] [answered]
Hi,

I am receiving the following error when I run v run main.v:

cannot import module "greeter" (not found)

#I am following along with the documentation, and I have the following two files:

src/
- main.v
- greeter.v

main.v:

```
module main
import greeter

fn main() {
	greeter.greet("Jason")
}
```

greeter.v:

```
module greeter

fn main() { }
pub fn greet(name string) {
	println(name)
}
```

Can somebody please help me understand where I'm going wrong? Even if I put the module into a modules folder, and then import it, I still get the same error.

**Comment by JalonSolov** [comment]
`greeter.v` should be in a subdirectory named `greeter` - that's how V defined modules.  It doesn't need a `fn main` unless you plan on running it separately.

You also need a `v.mod` file at the root of your project, as V used that as an anchor point when looking for local modules.  It can be an empty file, or you can create a full one as described in https://docs.vlang.io/package-management.html#publish-package

In other words, set things up like this for them to work:
```
.
├── src
│   ├── greeter
│   │   └── greeter.v
│   └── main.v
└── v.mod
```
main.v
```v
module main

import greeter

fn main() {
        greeter.greet('Jason')
}
```
greeter.v
```v
module greeter

pub fn greet(name string) {
        println(name)
}
```
output is as expected
```
$ v run .
Jason
$
```

**Reply by JalonSolov** [reply, marked as answer]
Last tip... the `src` subdir isn't required... some like to use it, some don't.  The code works exactly the same if you have
```
.
├── greeter
│   └── greeter.v
├── main.v
└── v.mod
```
```
$ v run .
Jason
$
```

**Reply by jason-pezzimenti** [reply]
Thanks.
